browser_utils.js

```diff
@@ -452,6 +452,95 @@ async function clickText(page, text, {timeout=30000, checkEvery=200, elementXPat
     });
 }
 
+/**
+ * Click any element by its text content.
+ * 
+ * The text can span multiple nodes compared to `clickText` which matches direct descended text nodes only.
+ * 
+ * @param {import('puppeteer').Page} page puppeteer page object.
+ * @param {string | RegExp} textOrRegExp Text or regex to match the text that the element must contain.
+ * @param {{extraMessage?: string}} [__namedParameters] Options (currently not visible in output due to typedoc bug)
+ * @param {string?} extraMessage Optional error message shown if the element is not visible in time.
+ */
+async function clickNestedText(page, textOrRegExp, {timeout=30000, checkEvery=200, extraMessage=undefined, visible=true}={}) {
+    if (typeof textOrRegExp === 'string') {
+        checkText(textOrRegExp);
+    }
+
+    const serializedMatcher = typeof textOrRegExp !== 'string' 
+        ? {source: textOrRegExp.source, flags: textOrRegExp.flags}
+        : textOrRegExp;
+
+    let remainingTimeout = timeout;
+    while (true) { // eslint-disable-line no-constant-condition
+        const found = await page.evaluate((matcher, visible) => {
+            // Optional: Deserialize jsonified RegExp object
+            const isStringMatcher = typeof matcher == 'string';
+            if (!isStringMatcher) {
+                matcher = new RegExp(matcher.source, matcher.flags);
+            }
+
+            const stack = [document.body];
+            let item;
+            let lastFound = null;
+            while (item = stack.pop()) { // eslint-disable-line no-cond-assign
+                // Optimization: If there is only one child we can immediately
+                // continue traversing and skip `.textContent` access.
+                if (item.childNodes.length === 1) {
```

Comment:
I'm not sure why this is an optimization in the first place, and why it is valid. Can we add a test case with `<html><body><button>clickme</button></body></html>`?

I believe that would not work.

Reply:
Now that I'm looking at it again with a pair of fresh eyes: You're completely right. It's the same logic as in the for loop an we can reuse that to take care of it.

Added the suggested test case :+1: